weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTable.tsx

```diff
@@ -388,13 +388,23 @@ export const CallsTable: FC<{
           }
           const op = operator ? operator : getDefaultOperatorForValue(value);
 
+          // All values added to the filter model should be strings, we
+          // only allow text-field input in the filter bar (even for numeric)
+          // They are converted during the backend mongo-style filter creation
+          let strVal: string;
+          if (typeof value !== 'string') {
+            strVal = JSON.stringify(value);
+          } else {
+            strVal = value;
+          }
+
           // Check if there is an exact match for field, operator, and value in filterModel.items
           // If an exact match exists, remove it instead of adding a duplicate.
           const existingFullMatchIndex = filterModel.items.findIndex(
             item =>
               item.field === field &&
               item.operator === op &&
-              JSON.stringify(item.value) === JSON.stringify(value)
+              JSON.stringify(item.value) === strVal
```

Comment:
I think you need to change this too, otherwise the additional quotes added on a string will prevent you from toggling it off.

Reply:
Good catch. fixed
![remove-filter-option](https://github.com/user-attachments/assets/337bdfa3-30cb-483a-aacd-5018e8b5ecb5)